### PR TITLE
[DEVX-1941] Add docs for playwright-cucumber

### PR DESCRIPTION
**Pull request by tianfeng92**
Add docs for playwright-cucumber test.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation fix (typos, incorrect content, missing content, etc.)

### Checklist

- [x] I have read the [contributing](https://github.com/saucelabs/sauce-docs/blob/main/docs/contributing.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: docs/web-apps/automated-testing/playwright/cucumber.md [hunk 1]
```diff
@@ -0,0 +1,870 @@
+---
+id: cucumber
+title: Configuring Your Playwright with Cucumber Tests
```
Comment (alexplischke) [inline docs/web-apps/automated-testing/playwright/cucumber.md:3]: I think we should change the general wording to `Cucumber Tests with Playwright`.
A benign change, but main focus are the Cucumber tests, with playwright as the automation framework underneath.

Also worth mentioning somewhere that this is CucumberJs, specifically.

File: docs/web-apps/automated-testing/_partials/_advanced-cucumber-playwright.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+## Running Cypress-js with Playwright
+
+`saucectl` provides a way to run Cucumber-js test with Playwright. In this case, Cucumber-js is running as the test runner and using Playwright as the test framework. You need to specified Cucumber-js related [configuration](/web-apps/automated-testing/playwright/cucumber).
```
Comment (alexplischke) [inline docs/web-apps/automated-testing/_partials/_advanced-cucumber-playwright.md:3]:
```suggestion
`saucectl` provides a way to run Cucumber.js tests with Playwright. In this case, Cucumber.js acts as the test framework, utilizing Playwright as the automation framework. This setup requires a dedicated Cucumber.js + Playwright [configuration](/web-apps/automated-testing/playwright/cucumber).
```

Comment (aliciakrebs) [inline docs/web-apps/automated-testing/_partials/_advanced-cucumber-playwright.md]:
```suggestion
`saucectl` provides a way to run Cucumber-js tests with Playwright. In this case, Cucumber-js is running as the test runner and using Playwright as the test framework. To run Cucumber-js tests with Playwright you'll need to specify the [Cucumber-js related configurations](/web-apps/automated-testing/playwright/cucumber).
```

File: docs/web-apps/automated-testing/playwright/cucumber.md [hunk 3]
```diff
@@ -0,0 +1,852 @@
...
+  options:
+    format:
+      - "json:my-cucumber.json"
+```
+---
+
+#### `formatOptions`
+<p><small>| OPTIONAL | OBJECT |</small></p>
+
+Options to be provided to formatters. You can check more details [here](https://github.com/cucumber/cucumber-js/blob/main/docs/formatters.md).
```
Comment (aliciakrebs) [inline docs/web-apps/automated-testing/playwright/cucumber.md:845]:
```suggestion
Options to provide to formatters. See the [Cucumber-js Formatters documentation](https://github.com/cucumber/cucumber-js/blob/main/docs/formatters.md) for more information. 
```

File: docs/web-apps/automated-testing/playwright/cucumber.md [hunk 4]
```diff
@@ -0,0 +1,852 @@
...
+    tags:
+      - "@smoke"
+      - "@e2e"
+```
+---
+
+#### `format`
+<p><small>| OPTIONAL | ARRAY |</small></p>
+
+Name/path and (optionally) output file path of each formatter to use. See more details [here](https://github.com/cucumber/cucumber-js/blob/main/docs/formatters.md).
```
Comment (aliciakrebs) [inline docs/web-apps/automated-testing/playwright/cucumber.md:833]:
```suggestion
Name/path and (optionally) output file path of each formatter to use. See the [Cucumber-js Formatters documentation](https://github.com/cucumber/cucumber-js/blob/main/docs/formatters.md) for more information.
```

File: docs/web-apps/automated-testing/playwright/cucumber.md [hunk 5]
```diff
@@ -0,0 +1,852 @@
...
+  options:
+    import:
+      - "features/support/*.js"
+```
+---
+
+#### `tags`
+<p><small>| OPTIONAL | ARRAY |</small></p>
+
+Tag expression to filter which Cucumber scenarios should be run. See more details [here](https://github.com/cucumber/cucumber-js/blob/main/docs/filtering.md#tags).
```
Comment (aliciakrebs) [inline docs/web-apps/automated-testing/playwright/cucumber.md:820]:
```suggestion
Tag expression to filter which Cucumber scenarios run. See the [Cucumber-js Filtering documentation](https://github.com/cucumber/cucumber-js/blob/main/docs/filtering.md#tags) for more information.
```

File: docs/web-apps/automated-testing/playwright/cucumber.md [hunk 6]
```diff
@@ -0,0 +1,852 @@
...
+  options:
+    require:
+      - "features/support/*.js"
+```
+---
+
+#### `import`
+<p><small>| OPTIONAL | ARRAY |</small></p>
+
+Paths to where your support code is, for ESM. See more details [here](https://github.com/cucumber/cucumber-js/blob/main/docs/esm.md).
```
Comment (aliciakrebs) [inline docs/web-apps/automated-testing/playwright/cucumber.md:808]:
```suggestion
Paths to your support code for ESM. See the [Cucumber-js ES Modules (experimental) documentation](https://github.com/cucumber/cucumber-js/blob/main/docs/esm.md) for more information. 
```

File: docs/web-apps/automated-testing/playwright/cucumber.md [hunk 7]
```diff
@@ -0,0 +1,852 @@
...
+```yaml
+  options:
+    backtrace: true
+```
+---
+
+#### `require`
+<p><small>| OPTIONAL | ARRAY |</small></p>
+
+Paths to where your support code is, for CommonJS. See more details [here](https://github.com/cucumber/cucumber-js/blob/main/docs/configuration.md#finding-your-code).
```
Comment (aliciakrebs) [inline docs/web-apps/automated-testing/playwright/cucumber.md:796]:
```suggestion
Paths to your support code for CommonJS. See the [Cucumber-js Configuration documentation](https://github.com/cucumber/cucumber-js/blob/main/docs/configuration.md#finding-your-code) for more information.
```

File: docs/web-apps/automated-testing/playwright/cucumber.md [hunk 8]
```diff
@@ -0,0 +1,852 @@
...
+```yaml
+  options:
+    excludedTestFiles: ["features/failed/**/*.feature"]
+```
+---
+
+#### `backtrace`
+<p><small>| OPTIONAL | BOOLEAN |</small></p>
+
+Shows the full backtrace for errors.
```
Comment (aliciakrebs) [inline docs/web-apps/automated-testing/playwright/cucumber.md:785]:
```suggestion
Specifies whether to show the full backtrace for errors.
```

File: docs/web-apps/automated-testing/playwright/cucumber.md [hunk 9]
```diff
@@ -0,0 +1,852 @@
...
+```yaml
+  options:
+    name: ".*My Cucumber Scenario"
+```
+---
+
+#### `paths`
+<p><small>| REQUIRED | ARRAY |</small></p>
+
+Paths to where the feature files are, using glob pattern. See more details [here](https://github.com/cucumber/cucumber-js/blob/main/docs/configuration.md#finding-your-features).
```
Comment (aliciakrebs) [inline docs/web-apps/automated-testing/playwright/cucumber.md:762]:
```suggestion
Paths to feature files. See the [Cucumber-js Configuration documentation](https://github.com/cucumber/cucumber-js/blob/main/docs/configuration.md#finding-your-features) for more information. You can use glob pattern to indicate all files that match a specific value, such as a file name, type, or directory.
```

File: docs/web-apps/automated-testing/playwright/cucumber.md [hunk 10]
```diff
@@ -0,0 +1,852 @@
...
+defaults:
+  mode: sauce
+  timeout: 15m
+```
+---
+
+### `mode`
+<p><small>| OPTIONAL | STRING/ENUM |</small></p>
+
+Instructs `saucectl` run tests remotely through Sauce Labs (`sauce`) or locally on `docker`. You can override this setting for individual suites using the `mode` setting within the [`suites`](#suites) object. If not set, the default value is `sauce`.
```
Comment (aliciakrebs) [inline docs/web-apps/automated-testing/playwright/cucumber.md:81]:
```suggestion
Instructs `saucectl` to run tests remotely through Sauce Labs (`sauce`) or locally on `docker`. You can override this setting for individual suites using the `mode` setting within the [`suites`](#suites) object. If not set, the default value is `sauce`.
```

File: docs/web-apps/automated-testing/playwright/cucumber.md [hunk 11]
```diff
@@ -0,0 +1,852 @@
...
+```
+---
+
+### `dependencies`
+<p><small>| OPTIONAL | ARRAY |</small></p>
+
+Specifies any npm packages that are required to run tests and should, therefore, be included in the bundle.
+The dependencies specified here have to be already installed in the local `node_modules` folder. These dependencies, along with any related transitive dependencies, are then included in the bundle that is uploaded to Sauce Labs.
+
+In order to run Cucumber-js with Playwright test, you need to install the following required packages locally and then add dependencies to the config.
```
Comment (aliciakrebs) [inline docs/web-apps/automated-testing/playwright/cucumber.md:355]:
```suggestion
In order to run Cucumber-js tests Playwright, you must to install the following required packages locally, and then add the dependencies to the configuration file.
```

File: docs/web-apps/automated-testing/playwright/cucumber.md [hunk 12]
```diff
@@ -0,0 +1,852 @@
...
+        - "features/support/*.js"
+      format:
+        - "json:my-cucumber.json"
+```
+---
+
+#### `name`
+<p><small>| OPTIONAL | STRING |</small></p>
+
+Regular expressions of which Cucumber scenario names should match one of to be run. See more details [here](https://github.com/cucumber/cucumber-js/blob/main/docs/filtering.md#names).
```
Comment (aliciakrebs) [inline docs/web-apps/automated-testing/playwright/cucumber.md:751]:
```suggestion
Specifies with regular expression matching which Cucumber scenarios to run. See the [Cucumber-js Filtering documentation](https://github.com/cucumber/cucumber-js/blob/main/docs/filtering.md#names) for more information.
```

File: docs/web-apps/automated-testing/playwright/cucumber.md [hunk 13]
```diff
@@ -0,0 +1,852 @@
...
+    - "@cucumber/cucumber"
+    - "@saucelabs/cucumber-reporter"
+    - "typescript"
+    - "ts-node"
+```
+
+To use this feature, make sure that `node_modules` is not ignored via `.sauceignore`.
+
+:::note
+`saucectl` doesn't support the way to run Cucumber-Playwright test via installing Cucumber-js related packages on the fly.
```
Comment (aliciakrebs) [inline docs/web-apps/automated-testing/playwright/cucumber.md:369]:
```suggestion
`saucectl` doesn't support running Playwright-Cucumber tests via installing Cucumber related packages on the fly.
```

**Review by aliciakrebs — CHANGES_REQUESTED** [review]

File: docs/web-apps/automated-testing/playwright/cucumber.md [hunk 14]
```diff
@@ -0,0 +1,852 @@
...
+    - "@cucumber/cucumber"
+    - "@saucelabs/cucumber-reporter"
+    - "typescript"
+    - "ts-node"
+```
+
+To use this feature, make sure that `node_modules` is not ignored via `.sauceignore`.
+
+:::note
+`saucectl` doesn't support running Playwright-Cucumber tests via installing Cucumber related packages on the fly.
```
Comment (aliciakrebs) [inline docs/web-apps/automated-testing/playwright/cucumber.md:369]:
```suggestion
`saucectl` doesn't support running Cucumber-js tests with Playwright via installing Cucumber related packages on the fly.
```

File: docs/web-apps/automated-testing/_partials/_advanced-cucumber-playwright.md [hunk 15]
```diff
@@ -0,0 +1,3 @@
+## Running Cypress-js with Playwright
+
+`saucectl` provides a way to run Cucumber.js tests with Playwright. In this case, Cucumber.js acts as the test framework, utilizing Playwright as the automation framework. This setup requires a dedicated Cucumber.js + Playwright [configuration](/web-apps/automated-testing/playwright/cucumber).
```
Comment (aliciakrebs) [inline docs/web-apps/automated-testing/_partials/_advanced-cucumber-playwright.md:3]:
```suggestion
`saucectl` provides a way to run Cucumber.js tests with Playwright. In this case, Cucumber.js acts as the test framework, utilizing Playwright as the automation framework. Running Cucumber-js tests with Playwright requires a dedicated  [configuration](/web-apps/automated-testing/playwright/cucumber).
```

File: docs/web-apps/automated-testing/playwright/cucumber.md [hunk 16]
```diff
@@ -0,0 +1,852 @@
...
+```
+---
+
+### `dependencies`
+<p><small>| OPTIONAL | ARRAY |</small></p>
+
+Specifies any npm packages that are required to run tests and should, therefore, be included in the bundle.
+The dependencies specified here have to be already installed in the local `node_modules` folder. These dependencies, along with any related transitive dependencies, are then included in the bundle that is uploaded to Sauce Labs.
+
+In order to run Cucumber-js tests Playwright, you must to install the following required packages locally, and then add the dependencies to the configuration file.
```
Comment (aliciakrebs) [inline docs/web-apps/automated-testing/playwright/cucumber.md:355]:
```suggestion
In order to run Cucumber-js tests with Playwright, you must to install the following required packages locally, and then add the dependencies to the configuration file.
```

File: docs/web-apps/automated-testing/playwright/cucumber.md [hunk 17]
```diff
@@ -0,0 +1,852 @@
...
+
+```yaml
+  - name: "saucy test"
+```
+### `browserName`
+<p><small>| OPTIONAL | STRING |</small></p>
+
+Sets the browser name for the test suite. `saucectl` passes `browserName` as an environment variable `$BROWSER_NAME`.
+
+Launching the browser for Cucumber-Playwright test should be done on customer's side. Hence `saucectl` cannot guarantee the displayed browser name is matched with the actual browser name.
```
Comment (aliciakrebs) [inline docs/web-apps/automated-testing/playwright/cucumber.md:627]:
```suggestion
Launching the browser for Cucumber.js Playwright tests should be done on customer's side. Hence `saucectl` cannot guarantee the displayed browser name is matched with the actual browser name.
```

**Review by aliciakrebs — CHANGES_REQUESTED** [review]

**Review by lysannep — APPROVED** [review]

**Review by aliciakrebs — APPROVED** [review]

**Pull request merged**